kong/cmd/utils/prefix_handler.lua

```diff
@@ -93,17 +93,24 @@ local function find_resty_bin()
   return found
 end
 
-local function gen_default_ssl_cert(kong_config)
+local function gen_default_ssl_cert(kong_config, admin)
   -- create SSL folder
   local ok, err = pl_dir.makepath(pl_path.join(kong_config.prefix, "ssl"))
   if not ok then return nil, err end
 
-  local ssl_cert = kong_config.ssl_cert_default
-  local ssl_cert_key = kong_config.ssl_cert_key_default
-  local ssl_cert_csr = kong_config.ssl_cert_csr_default
+  local ssl_cert, ssl_cert_key, ssl_cert_csr
+  if admin then
+    ssl_cert = kong_config.admin_ssl_cert_default
+    ssl_cert_key = kong_config.admin_ssl_cert_key_default
+    ssl_cert_csr = kong_config.admin_ssl_cert_csr_default
+  else
+    ssl_cert = kong_config.ssl_cert_default
+    ssl_cert_key = kong_config.ssl_cert_key_default
+    ssl_cert_csr = kong_config.ssl_cert_csr_default
+  end
 
   if not pl_path.exists(ssl_cert) and not pl_path.exists(ssl_cert_key) then
-    log.verbose("generating default SSL certificate and key")
+    log.verbose("generating "..(admin and "admin" or "default").." SSL certificate and key")
```

Comment:
Let's make this more concise:
```lua
log.verbose("generating %s SSL certificate and key",
                     admin and "admin" or "default") -- notice the line jump to avoid >80 cols
```